common/datastore/conversions.go

```diff
@@ -59,9 +59,10 @@ func toObject(raw_data interface{}, definition DataStoreDefinition) (interface{}
 			if err != nil {
 				return nil, NewErrInField(field.Name, err)
 			}
-		} else {
-			data[field.Name] = getDefaultValue(field.Type)
-		}
+		} 
+		// else {
+		// 	data[field.Name] = getDefaultValue(field.Type) //removed for registration/attendee patch function
+		// }
```

Comment:
Hm. Now that I think about it, I think we actually don't want to do this. Users currently expect the API to always return all fields, even if their value is empty/default. 1. We wouldn't want to introduce a change to this behavior without thinking about its implications on the user and 2. I think the API having a consistent return format is better.

So as for how to get around this issue: we can try to manually strip off fields from the user_registration when we pass it into the database patch function